### ISSUE TITLE
Include video size and normalisation information in model metadata

### ISSUE THREAD

**Issue by fededagos** [open-retina/open-retina#89]
> Works for all pretrained models now. Example command:
> 
> ```
> ./scripts/visualize_model_neurons.py --model_path models/hoefling_2024_GRU_high_res_model.ckpt --save_folder vis/hoefling_2024_GRU_high_res_model
> ```
> 
> Changes:
> 
>     * Make plotting and video generation work for grey-scale videos
> 
>     * Optionally infer stimulus shape from filename [1]
> 
>     * Update visualize_model_neurons script to work with all models
> 
> 
> [1] It might be useful to store the input shape in the CoreReadoutModel, too (we currently pass it to the constructor, but do not store it their). If doing so, we would not have to pass it to the visualize_model_neurons script.

Regarding [1]: I was thinking about this already regarding the video normalisation, and now this is one more strong reason to include input information directly in the model metadata. On top of this, do you think we should include also the values of the video used in normalisation? I noticed they are used in a couple of functions in the script.
And should we make these changes now or incorporate this fix for the time being and deal with these problems later?

_Originally posted by @fededagos in https://github.com/open-retina/open-retina/issues/87#issuecomment-2587350963_

---

The following feature is necessary for several reasons:
1. Easier access to video size metadata for MEI generation
2. Consistent normalisation of models trained on more than one video → video normalisation mean and std need to be made hyperparameters and stored inside the model
3. Easier (and necessary) access to video normalisation values when evaluating the model on an unseen clip (of which the normalisation should follow what was used in training)

**Comment by thomasZen** [comment]
Maybe also think about adding additional information useful for downstream analysis in the model (e.g. celltype for each output neuron, maybe receptive field)
